sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/_evaluators/_coherence/_coherence.py

```diff
@@ -49,6 +49,42 @@ def __init__(self, model_config):
         prompty_path = os.path.join(current_dir, self._PROMPTY_FILE)
         super().__init__(model_config=model_config, prompty_file=prompty_path, result_key=self._RESULT_KEY)
 
+    @overload
+    def __call__(
+        self,
+        *,
+        query: str,
+        response: str,
+    ):
```

Comment:
Return type for `__call__` is not a thing?

Reply:
we can't use `typing.Dict` since `PFClient.save()` rejects it as a complex type. I have a task to fix it, but haven't had a chance to make the fix yet.

The API view picks up the type hints from the docstrings if it's missing, so from Azure SDK perspective, we're good.

Reply:
nvm on this actually. Promptflow does `inspect.signature`  when saving, which only picks up the non-overloaded function.. meaning the overloaded ones CAN have richer typing